### PR TITLE
Model test cases

### PR DESCRIPTION
**Pull request by namrataukirde**

### CODE REVIEW

File: tests/unit/models/item-test.js [hunk 1]
```diff
@@ -59,3 +59,102 @@ test('allPackageCategories', function(assert) {
 
   assert.equal(record.get('allPackageCategories.firstObject.id'), [pkgCategory.id]);
 });
+
+
+test('isItem', function(assert){
```
Comment (Darshan-Chauhan) [inline tests/unit/models/item-test.js]: Can we give some proper description for the test?

File: tests/unit/models/item-test.js [hunk 2]
```diff
@@ -59,3 +59,102 @@ test('allPackageCategories', function(assert) {
...
+  var record, pkgType;
+  var subject = this.subject();
+  assert.expect(1);
+
+  Ember.run(function() {
+    subject.store.createRecord('package_type', {id: 5, code: "ABC"});
```
Comment (Darshan-Chauhan) [inline tests/unit/models/item-test.js]: Can we do **pkgType = subject.store.createRecord('package_type', {id: 5, code: "ABC"});**?

File: tests/unit/models/item-test.js [hunk 3]
```diff
@@ -59,3 +59,102 @@ test('allPackageCategories', function(assert) {
...
+    pkgType = subject.store.peekRecord('package_type', 5);
+
+    subject.store.createRecord('item', {id: 5, packageType: pkgType});
+    record = subject.store.peekRecord('item', 5);
```
Comment (Darshan-Chauhan) [inline tests/unit/models/item-test.js]: ^

File: tests/unit/models/item-test.js [hunk 4]
```diff
@@ -59,3 +59,102 @@ test('allPackageCategories', function(assert) {
...
+  });
+
+  assert.equal(record.get('isItem'), true);
+});
+
+
+test('isAvailable', function(assert){
```
Comment (Darshan-Chauhan) [inline tests/unit/models/item-test.js]: Can we give some proper description for the test?

File: tests/unit/models/item-test.js [hunk 5]
```diff
@@ -59,3 +59,102 @@ test('allPackageCategories', function(assert) {
...
+  Ember.run(function(){
+    package1 = store.createRecord('package', { id: 1, quantity: 2, isAvailable: true });
+    package2 = store.createRecord('package', { id: 2, quantity: 2, isAvailable: true });
+    model.get('packages').pushObjects([package1, package2]);
+  });
+
+  assert.equal(model.get('isAvailable'), true);
+});
+
+test('quantity', function(assert){
```
Comment (Darshan-Chauhan) [inline tests/unit/models/item-test.js]: ^

File: tests/unit/models/item-test.js [hunk 6]
```diff
@@ -59,3 +59,102 @@ test('allPackageCategories', function(assert) {
...
+  Ember.run(function(){
+    package1 = store.createRecord('package', { id: 1, quantity: 2, isUnavailableAndDesignated: true });
+    package2 = store.createRecord('package', { id: 2, quantity: 2, isUnavailableAndDesignated: false });
+    model.get('packages').pushObjects([package1, package2]);
+  });
+
+  assert.equal(model.get('isUnavailableAndDesignated'), false);
+});
+
+test('images', function(assert){
```
Comment (Darshan-Chauhan) [inline tests/unit/models/item-test.js]: ^

File: tests/unit/models/item-test.js (hunk 1 above)
Comment (namrataukirde) [reply tests/unit/models/item-test.js]: Sure 👍

**Pull request merged**